src/training_raster_clipper/implementation/nbertaud.py

```diff
@@ -0,0 +1,194 @@
+from pathlib import Path
+
+import geopandas.io.file
+import numpy as np
+import pandas as pd
+import rasterio.features
+import rioxarray
+import xarray as xr
+from geopandas.geodataframe import GeoDataFrame
+from sklearn.ensemble import RandomForestClassifier
+import joblib
+
+from training_raster_clipper.custom_types import (
+    BandNameType,
+    ClassificationResult,
+    ClassifiedSamples,
+    FeatureClassNameToId,
+    PolygonMask,
+    ResolutionType,
+)
+
+
+def load_feature_polygons(input_path: Path) -> GeoDataFrame:
+    gdf = geopandas.io.file.read_file(input_path)
+    gdf = gdf.to_crs(32631)
+    return gdf
+
+
+def load_sentinel_data(
+    sentinel_product_location: Path,
+    resolution: ResolutionType,
+    band_names: tuple[BandNameType, ...],
+) -> xr.DataArray:
+    """Loads sentinel product
+
+    Example input path: `S2A_MSIL2A_20221116T105321_N0400_R051_T31TCJ_20221116T170958.SAFE`
+
+    Args:
+        sentinel_product_location (Path): Location of the .SAFE folder containing a Sentinel-2 product.
+
+    Returns:
+        xr.DataArray: A DataArray containing the 3 RGB bands from the visible spectrum
+    """
+
+    # Find the path of the folder with the correct resolution
+    spatial_resolution_location = list(sentinel_product_location.glob(f"GRANULE/*/IMG_DATA/R{resolution}m"))[0]
+
+    # Find the path of the bands
+    band_locations = {}
+    for band_name in band_names:
+        band_locations[band_name] = list(spatial_resolution_location.glob(f"*_{band_name}_*"))[0]
```

Comment:
you can do it in one expression with  a dict comprehension eg

instead of A (imperative)

```python
band_locations = {}
for band_name in band_names:
    d[band_name] = f(band_name)
```

you can do B (declarative)

```python
band_locations = { band_name: f(band_name) for band_name in band_names }
```